lib/foreman_maintain.rb

```diff
@@ -28,8 +29,19 @@ module ForemanMaintain
   class << self
     attr_accessor :config, :logger
 
-    def setup(configuration = {})
-      self.config = Config.new(configuration)
+    LOGGER_LEVEL_MAPPING = {
+      'debug' => ::Logger::DEBUG,
+      'info' => ::Logger::INFO,
+      'warn' => ::Logger::WARN,
+      'error' => ::Logger::ERROR,
+      'fatal' => ::Logger::FATAL,
+      'unknown' => ::Logger::UNKNOWN
+    }.freeze
+
+    def setup(options = {})
+      @msg_queue = []
```

Comment:
The `@msg_queue` could have a more descriptive name. What about `@pre_setup_log_messages`, and + a small comment why we are doing so.